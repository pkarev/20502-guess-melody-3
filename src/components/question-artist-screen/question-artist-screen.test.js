import React from 'react';
import renderer from 'react-test-renderer';
import QuestionArtistScreen from './question-artist-screen';
import withActivePlayer from '../../hocs/with-active-player/with-active-player';

const QuestionArtistScreenWrapped = withActivePlayer(QuestionArtistScreen);

const questionArtist = {
  artist: `Пелагея`,
  src: `https://upload.wikimedia.org/wikipedia/commons/4/4e/BWV_543-fugue.ogg`,
  options: [
    {
      artist: `Пелагея`,
      picture: `https://htmlacademy-react-3.appspot.com/guess-melody/static/artist/Quincas_Moreira.jpg`,
    },
    {
      artist: `Меладзе`,
      picture: `https://htmlacademy-react-3.appspot.com/guess-melody/static/artist/Jesse_Gallagher.jpg`,
    },
    {
      artist: `Шнуров`,
      picture: `https://htmlacademy-react-3.appspot.com/guess-melody/static/artist/sextile.jpg`,
    },
  ]
};

it(`Render QuestionArtistScreen`, () => {
  const tree = renderer
    .create(<QuestionArtistScreenWrapped
      question={questionArtist}
      onAnswer={() => {}}
    />, {
      createNodeMock: () => {
        return {};
      }
    })
    .toJSON();

  expect(tree).toMatchSnapshot();
});
