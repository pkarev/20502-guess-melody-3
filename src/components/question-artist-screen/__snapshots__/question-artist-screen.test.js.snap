// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render QuestionArtistScreen 1`] = `
<section
  className="game__screen"
>
  <h2
    className="game__title"
  >
    Кто исполняет эту песню?
  </h2>
  <div
    className="game__track"
  >
    <div
      className="track"
    >
      <button
        className="track__button track__button--play"
        disabled={true}
        onClick={[Function]}
        type="button"
      />
      <div
        className="track__status"
      >
        <audio />
      </div>
    </div>
  </div>
  <form
    className="game__artist"
  >
    <div
      className="artist"
    >
      <input
        className="artist__input visually-hidden"
        id="answer-0"
        name="answer"
        onChange={[Function]}
        type="radio"
        value="answer-0"
      />
      <label
        className="artist__name"
        htmlFor="answer-0"
      >
        <img
          alt="Пелагея"
          className="artist__picture"
          src="https://htmlacademy-react-3.appspot.com/guess-melody/static/artist/Quincas_Moreira.jpg"
        />
        Пелагея
      </label>
    </div>
    <div
      className="artist"
    >
      <input
        className="artist__input visually-hidden"
        id="answer-1"
        name="answer"
        onChange={[Function]}
        type="radio"
        value="answer-1"
      />
      <label
        className="artist__name"
        htmlFor="answer-1"
      >
        <img
          alt="Меладзе"
          className="artist__picture"
          src="https://htmlacademy-react-3.appspot.com/guess-melody/static/artist/Jesse_Gallagher.jpg"
        />
        Меладзе
      </label>
    </div>
    <div
      className="artist"
    >
      <input
        className="artist__input visually-hidden"
        id="answer-2"
        name="answer"
        onChange={[Function]}
        type="radio"
        value="answer-2"
      />
      <label
        className="artist__name"
        htmlFor="answer-2"
      >
        <img
          alt="Шнуров"
          className="artist__picture"
          src="https://htmlacademy-react-3.appspot.com/guess-melody/static/artist/sextile.jpg"
        />
        Шнуров
      </label>
    </div>
  </form>
</section>
`;
